src/Client/ConnectionParameters.cpp

```diff
@@ -57,6 +57,14 @@ ConnectionParameters::ConnectionParameters(const Poco::Util::AbstractConfigurati
             password = result;
     }
 
+    const char * env_user = getenv("CLICKHOUSE_USER");
+    const char * env_password = getenv("CLICKHOUSE_PASSWORD");
+    if (!config.has("user") && !config.has("password") && env_user && env_password)
```

Comment:
User may want to set just `CLICKHOUSE_USER` or `CLICKHOUSE_PASSWORD` but not both, so we can set it separately

Reply:
Also values env should have higher priority than config, so set it even if config have some setup

Reply:
So about priority, command user >  environment variables > config.xml config

Reply:
> So about priority, command user > environment variables > config.xml config

Yes